options/options.go

```diff
@@ -276,9 +276,27 @@ func (terragruntOptions *TerragruntOptions) InsertTerraformCliArgs(argsToInsert
 	// Options must be inserted after command but before the other args
 	// command is either 1 word or 2 words
 	var args []string
+	var lastArg string
+
 	args = append(args, terragruntOptions.TerraformCliArgs[:commandLength]...)
-	args = append(args, argsToInsert...)
+	// We need to iterate over the argsToInsert to find an arg that is a file.
+	// If we found one, we treat it as the Terraform plan file and we then set it
+	// as lastArg to append it as last argument.
```

Comment:
Hm, I'm not sure this is a safe assumption to make. That is, it's possible to have a different arg passed to Terraform that is a file—or worse yet, happens to match the name of a valid file—but isn't a plan file. Here's an example:

```bash
terraform apply \ 
  -state foo.tfstate \
  -var foo=bar
```

If there's a `foo.tfstate` already, the code above will move that parameter to the end of the command, as if it's a plan file.

Reply:
Thank you @brikis98, I haven't considered that use case. I will write a test as well as validating the current ones work fine and come back to you.